ffi/capi/src/iana_parser.rs

```diff
@@ -47,149 +44,19 @@ pub mod ffi {
             FnInStruct,
             hidden
         )]
-        pub fn iana_to_bcp47(
-            &self,
-            value: &DiplomatStr,
-            write: &mut diplomat_runtime::DiplomatWrite,
-        ) {
-            let handle = self.0.as_borrowed();
-            let bcp47 = handle.iana_bytes_to_bcp47(value);
-            let _infallible = bcp47.0.write_to(write);
-        }
-
-        #[diplomat::rust_link(
-            icu::time::zone::iana::IanaParserBorrowed::normalize_iana,
-            FnInStruct
-        )]
-        pub fn normalize_iana(
-            &self,
-            value: &str,
-            write: &mut diplomat_runtime::DiplomatWrite,
-        ) -> Option<()> {
-            let handle = self.0.as_borrowed();
-            let iana = handle.normalize_iana(value)?;
-            let _infallible = iana.0.write_to(write);
-            Some(())
-        }
-
-        #[diplomat::rust_link(
-            icu::time::zone::iana::IanaParserBorrowed::canonicalize_iana,
-            FnInStruct
-        )]
-        pub fn canonicalize_iana(
-            &self,
-            value: &str,
-            write: &mut diplomat_runtime::DiplomatWrite,
-        ) -> Option<()> {
-            let handle = self.0.as_borrowed();
-            let iana = handle.canonicalize_iana(value)?;
-            let _infallible = iana.0.write_to(write);
-            Some(())
-        }
-
-        #[diplomat::rust_link(
-            icu::time::zone::iana::IanaParserBorrowed::find_canonical_iana_from_bcp47,
-            FnInStruct
-        )]
-        pub fn find_canonical_iana_from_bcp47(
-            &self,
-            value: &DiplomatStr,
-            write: &mut diplomat_runtime::DiplomatWrite,
-        ) -> Option<()> {
-            let handle = self.0.as_borrowed();
-            let iana = TinyAsciiStr::try_from_utf8(value)
-                .ok()
-                .and_then(|s| handle.find_canonical_iana_from_bcp47(icu_time::TimeZone(s)))?;
-            let _infallible = iana.write_to(write);
-            Some(())
-        }
-    }
-
-    /// A mapper between IANA time zone identifiers and BCP-47 time zone identifiers.
-    ///
-    /// This mapper supports two-way mapping, but it is optimized for the case of IANA to BCP-47.
-    /// It also supports normalizing and canonicalizing the IANA strings.
-    #[diplomat::opaque]
-    #[diplomat::rust_link(icu::time::zone::iana::IanaParserExtended, Struct)]
-    #[diplomat::rust_link(
-        icu::time::zone::iana::IanaParserExtended::as_borrowed,
-        FnInStruct,
-        hidden
-    )]
-    #[diplomat::rust_link(icu::time::zone::iana::IanaParserExtended::inner, FnInStruct, hidden)]
-    #[diplomat::rust_link(icu::time::zone::iana::IanaParserExtendedBorrowed, Struct, hidden)]
-    #[diplomat::rust_link(
-        icu::time::zone::iana::IanaParserExtendedBorrowed::inner,
-        FnInStruct,
-        hidden
-    )]
-    pub struct IanaParserExtended(
```

Comment:
I had to remove `IanaParserExtended` for now because we cannot return `(TimeZone, &str, &str)` from Diplomat. Need some API design.